conftest.py

```diff
@@ -13,3 +17,66 @@ def record_test_timestamp(record_property):
     yield
     end_timestamp = datetime.strftime(datetime.now(), "%Y-%m-%dT%H:%M:%S%z")
     record_property("end_timestamp", end_timestamp)
+
+
+@pytest.fixture(autouse=True)
+def memory_usage_tracker():
+    """
+    A pytest fixture that tracks memory usage during the execution of a test.
+
+    This fixture automatically tracks the memory usage of the process running the tests.
+    It starts tracking before the test runs, continues tracking in a background thread during the test,
+    and stops tracking after the test completes. It logs the memory usage statistics including the
+    minimum, maximum, average, and total memory usage by the test.
+
+    The memory usage is measured in megabytes (MB).
+
+    Note:
+        - This fixture is automatically used for all tests due to the `autouse=True` parameter.
+        - The interval for memory readings can be adjusted by changing the sleep duration in the `track_memory` function.
+        - Min, max, and avg memory usage are calculated based on the recorded memory readings from system memory.
+    """
+    process = psutil.Process()
+    memory_readings = []
+
+    # Capture start memory and initialize tracking
+    start_mem = process.memory_info().rss / (1024 * 1024)  # MB
+    memory_readings.append(start_mem)
+
+    # Start a background thread or loop to collect memory usage over time
+    tracking = True
+
+    def track_memory():
+        while tracking:
+            current_mem = process.memory_info().rss / (1024 * 1024)
```

Comment:
let's aggregate in place (not accumulating the whole list); don't want this to create a memory overflow 😄